packages/fast-components-styles-msft/src/utilities/high-contrast.ts

```diff
@@ -0,0 +1,219 @@
+import { CSSRules } from "@microsoft/fast-jss-manager";
+import { DesignSystem } from "../design-system";
+import { format, toPx } from "@microsoft/fast-jss-utilities";
+import { focusOutlineWidth, outlineWidth } from "./design-system";
+
+enum HighContrastColor {
+    text = "WindowText",
+    hyperLinks = "LinkText",
+    disabledText = "GrayText",
+    selectedText = "HighlightText",
+    selectedBackground = "Highlight",
+    buttonText = "ButtonText",
+    buttonBackground = "ButtonFace",
+    background = "Background",
+}
+
+export const highContrastSelector: string = "@media (-ms-high-contrast:active)";
+
+export function applyHighContrastAdjustStealth(): CSSRules<DesignSystem> {
+    return {
+        [highContrastSelector]: {
+            background: HighContrastColor.buttonBackground,
+            color: HighContrastColor.buttonText,
+            fill: HighContrastColor.buttonText,
+            border: "none",
+            "-ms-high-contrast-adjust": "none",
+        },
+    };
+}
+
+export function applyHighContrastAdjustOutline(): CSSRules<DesignSystem> {
+    return {
+        [highContrastSelector]: {
+            background: HighContrastColor.buttonBackground,
+            borderColor: HighContrastColor.buttonText,
+            color: HighContrastColor.buttonText,
+            fill: HighContrastColor.buttonText,
+            "-ms-high-contrast-adjust": "none",
+        },
+    };
+}
+
+export function applyHighContrastBorder(): CSSRules<DesignSystem> {
+    return {
+        [highContrastSelector]: {
+            background: HighContrastColor.buttonBackground,
+            border: format("{0} solid ButtonText", toPx<DesignSystem>(outlineWidth)),
+        },
+    };
+}
+
+export function applyHighContrastAdjustSelectOutline(): CSSRules<DesignSystem> {
```

Comment:
I think you can drop "Adjust" as that's an implementation detail. Maybe standardize on "Selected" (+ed). I think you can drop "Outline" as well as this just happens to be the selected state, for anything whether it has a border or not.

Reply:
updated to applyHighContrastSelected